### PR TITLE
Fixes #14: Parsing nested files for events and functions

### LINKED ISSUE #14: Implement external file references to the Workflow parser
**What would you like to be added**:
In the specification, we have the possibility to add external files references in the workflow definition, for instance:

```json
{
  "id": "VetAppointmentWorkflow",
  "name": "Vet Appointment Workflow",
  "description": "Vet service call via events",
  "version": "1.0",
  "events": "features/vetappointmenteventrefs.json",
  "retries": "features/vetappointmentretries.json",
  "states": [
    {
      "name": "MakeVetAppointmentState",
      "type": "operation",
      "start": {
        "kind": "default"
      },
      "actions": [
        {
          "name": "MakeAppointmentAction",
          "eventRef": {
            "triggerEventRef": "MakeVetAppointment",
            "data": "{{ $.patientInfo }}",
            "resultEventRef":  "VetAppointmentInfo"
          },
          "actionDataFilter": {
            "dataResultsPath": "{{ $.appointmentInfo }}"
          },
          "timeout": "PT15M"
        }
      ],
      "onErrors": [
        {
          "error": "TimeoutError",
          "code": "500",
          "retryRef": "TimeoutRetryStrategy",
          "end": {
            "kind": "default"
          }
        }
      ],
      "end": {
        "kind": "default"
      }
    }
  ]
}
```

In this example, `retries` and `events` are defined externally to the main file. The SDK supposed to incorporate those while parsing the main definition.

**Why is this needed**:
To conform with the specification.

### PR DESCRIPTION
**Pull request by iwita**
**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:
In this PR,  the functionality of parsing nested files for events and functions has been added. 

**Special notes for reviewers**:
I added a few more tests for: 
- Reading nested file from url
- Reading nested file from relative path
- Reading nested file from absolute path (in this case I only added the file:// prefix to the relative path, since it depends to the current dir of the user.

**Additional information (if needed):**

### CODE REVIEW

File: model/unmarshal_json.go [hunk 1]
```diff
@@ -15,10 +15,38 @@
...
+	if resp, err := http.Get(s); err == nil {
+		defer resp.Body.Close()
+		buf := new(bytes.Buffer)
+		if _, err = buf.ReadFrom(resp.Body); err != nil {
+			return nil, err
+		}
+		return buf.Bytes(), nil
+	}
+	if strings.HasPrefix(s, "file://") {
+		s = strings.TrimPrefix(s, "file:/")
```
Comment (ricardozanini) [inline model/unmarshal_json.go:38]: Can you please add those to a `const`?

**Review by ricardozanini — COMMENTED** [review]
Great work, @iwita ! Just one minor thing regarding a constant. Also, can you move the data from `features` package to `testdata` instead? Since these resources are only used in the test cases, I think would be better to leave them there.

Nice test cases btw. Many thanks for this PR!!

File: model/unmarshal_json.go (hunk 1 above)
Comment (iwita) [reply model/unmarshal_json.go:38]: Do you mean the prefix "file://"?

**Comment by iwita** [comment]
Thank you @ricardozanini !
I will add your additional comments.

> Nice test cases btw. Many thanks for this PR!!

I had created another test case for the `file://` prefix but I was able only to test it in my working directory

File: model/unmarshal_json.go (hunk 1 above)
Comment (ricardozanini) [reply model/unmarshal_json.go:38]: Yes!

**Comment by ricardozanini** [comment]
> I had created another test case for the file:// prefix but I was able only to test it in my working directory

np!

> Shoud I follow this pattern in files structure? https://github.com/serverlessworkflow/sdk-java/tree/master/api/src/test/resources

Nops. In our case, `testdata` is the directory to put resources that we only use on test cases. So the files you've added to the `features` directory actually should be in `testdata`. In the files you refer to them just change from `"functions": "features/applicationrequestfunctions.json",` to ` "functions": "testdata/applicationrequestfunctions.json",`. Makes sense?

**Comment by iwita** [comment]
> Nops. In our case, testdata is the directory to put resources that we only use on test cases. So the files you've added to the features directory actually should be in testdata. In the files you refer to them just change from "functions": "features/applicationrequestfunctions.json", to "functions": "testdata/applicationrequestfunctions.json",. Makes sense?

Yeah, sure. Just double checked it.

**Review by ricardozanini — APPROVED** [review]

File: hack/generate-spec-types.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ declare package="model"
 declare targetdir="/tmp/serverlessworkflow"
 
 if [ ! -d "${targetdir}" ]; then
-  git clone git@github.com:serverlessworkflow/specification.git ${targetdir}
+  git clone --single-branch --branch fix-zeros git@github.com:ricardozanini/sw-specification.git ${targetdir}
```
Comment (ricardozanini) [inline hack/generate-spec-types.sh:25]: I believe we need to revert this :)

**Review by ricardozanini — CHANGES_REQUESTED** [review]
There's just a script change that we need to revert

File: hack/generate-spec-types.sh (hunk 2 above)
Comment (iwita) [reply hack/generate-spec-types.sh]: yes, my bad :(

**Comment by ricardozanini** [comment]
@iwita were you able to run `make addheaders`?

**Comment by iwita** [comment]
Hi,
> @iwita were you able to run make addheaders?

I just submitted a new commit:
Changed the path of addlicense and clean the local environment afterwards.
However, while this change executes the Makefile successfully locally, it does not pass the automated tests here.

**Comment by ricardozanini** [comment]
> Hi,
> 
> > @iwita were you able to run make addheaders?
> 
> I just submitted a new commit:
> Changed the path of addlicense and cleaned the local environment afterwards.
> However, while this change executes the Makefile successfully locally, it does not pass the automated tests here.

That's because `GOPATH` is not part of this step in the GH Actions. Can you give a try to this #20? I believe that now you should be able to run the tooling in your local env.

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**